sensor/kubernetes/complianceoperator/handler_impl.go

```diff
@@ -241,7 +255,82 @@ func (m *handlerImpl) processRerunScheduledScanRequest(requestID string, request
 	}
 	return m.composeAndSendApplyScanConfigResponse(requestID, err)
 }
+func (m *handlerImpl) processScanConfigScheduleChangeRequest(requestID string, config scanScheduleConfiguration) bool {
+	if err := config.ValidationFunc(config.Request); err != nil {
+		return m.composeAndSendApplyScanConfigResponse(requestID, errors.Wrap(err, "validating compliance scan request"))
+	}
+
+	ns := m.complianceOperatorInfo.GetNamespace()
+	if ns == "" {
+		return m.composeAndSendApplyScanConfigResponse(requestID, errors.New("Compliance operator namespace not known"))
+	}
+
+	resI := m.client.Resource(complianceoperator.ScanSetting.GroupVersionResource()).Namespace(ns)
+	obj, err := resI.Get(m.ctx(), config.ScanName, v1.GetOptions{})
+	if err != nil || obj == nil {
+		err = errors.Wrapf(err, "namespaces/%s/scansettings/%s not found", ns, config.ScanName)
+		return m.composeAndSendApplyScanConfigResponse(requestID, err)
+	}
+
+	var scanSetting v1alpha1.ScanSetting
+	if err := runtime.DefaultUnstructuredConverter.FromUnstructured(obj.Object, &scanSetting); err != nil {
+		err = errors.Wrap(err, "Could not convert unstructured to scan setting")
+		return m.composeAndSendApplyScanConfigResponse(requestID, err)
+	}
+
+	// Check if scanSetting has the "Suspend" field
+	if _, ok := reflect.TypeOf(scanSetting).FieldByName("Suspend"); ok {
+		scanSetting.Suspend = *config.Suspend
+	} else {
+		// Handle the case where the field doesn't exist (older CRD)
+		return m.composeAndSendApplyScanConfigResponse(requestID, errors.New("suspending a scan is not supported on this version of the compliance operator"))
```

Comment:
This reminds me - where we going to have a check somewhere that ensured the Compliance Operator was at least version 1.4.0 for the integration to work?

Reply:
https://issues.redhat.com/browse/ROX-18101

It wasn't so much that it was on 1.4.0 but that they were all the same.  Probably worth re-visiting that one in the relatively near future.